### PR TITLE
operator/ingress: Fix up HTTP cookie capture API

### PR DESCRIPTION
**Pull request by Miciah**
Relax validation on `spec.logging.access.httpCaptureCookies` to allow the `name` and `namePrefix` subfields to be empty.

* `operator/v1/types_ingress.go` (`IngressControllerCaptureHTTPCookie`): Relax validation by allowing `Name` and `NamePrefix` to be empty.
* `operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml-merge-patch`: Delete validation added in #687.
* `operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml`:
* `operator/v1/zz_generated.swagger_doc_generated.go`: Regenerate.

----

Turns out what worked with `oc` commands does not work with client-go.  Specifying `matchType: Exact` and `name: foo` worked with `oc apply` but failed when updating using client-go:

    IngressController.operator.openshift.io "cookiecapture" is invalid: spec.logging.access.httpCaptureCookies.namePrefix: Invalid value: "": spec.logging.access.httpCaptureCookies.namePrefix in body should be at least 1 chars long

### CODE REVIEW

File: operator/v1/types_ingress.go [hunk 1]
```diff
@@ -672,25 +672,27 @@ type IngressControllerCaptureHTTPCookie struct {
...
+	// +kubebuilder:validation:MinLength=0
 	// +kubebuilder:validation:MaxLength=1024
 	// +optional
 	NamePrefix string `json:"namePrefix"`
 
-	// maxLength specifies a maximum length for the cookie value.  If a
-	// cookie value exceeds this length, the value will be truncated in the
-	// log message.  Note that the ingress controller may impose a separate
-	// bound on the total length of HTTP headers in a request.
+	// maxLength specifies a maximum length of the string that will be
```
Comment (deads2k) [inline operator/v1/types_ingress.go:690]: actually, why is this required.  Seems like we could choose a reasonable default max length

Comment (Miciah) [reply operator/v1/types_ingress.go:690]: I guess we could, but it would be pretty arbitrary because we do not know the cookie's purpose.  Would 256 be a reasonable default?

Comment (Miciah) [reply operator/v1/types_ingress.go:690]: Does the minimum need to be set to 0 in order for validation to admit an object that does not specify the value?
